### ISSUE TITLE
How can I open all parent nodes of a specified node?

### ISSUE THREAD

**Issue by srkunze** [vakata/jstree#1459]
Background: I wrote a search bar which selects the first found node. After clearing the search the node should stay selected, the tree should be opened to the specified node (i.e. all parents of this node) and the tree should be scrolled to the node.

Scrolling and selecting works.

However, the tree cannot be opened to the selected node. I tried:

1) select_node
2) show_node
3) _open_to

All three did not open all parent nodes of the given node. What can I do here?

PS: Scrolling does also not work when all parents are opened. The corresponding DOM elements seems to be undefined in this case.

**Comment by srkunze** [comment]
PS: we have 'search': {show_only_matches: true, show_only_matches_children: true} as config for the search plugin

**Comment by vakata** [comment]
Searching (using the `search()` jstree function) will automatically open all nodes to reveal the search matches. My guess is you are either not using the `search` function or not using the `search.jstree` event to execute your code.

If you do not want to use the `search()` function then you are probably dealing with a lazy loaded tree where the node you are searching for is not yet loaded. Also make sure you use IDs and not selectors (do NOT call `select_node('#node_id')`, DO call `select_node('node_id')`).

For further how-to questions please use the discussion group.

Best regards,
Ivan

**Comment by srkunze** [comment]
> Searching (using the search() jstree function) will automatically open all nodes to reveal the search matches.

Maybe, I didn't express myself well enough. When searching (using the `search` function AND executing my code using the `search.jstree` event), everything works as expected. (I also have a version where the code is executed directly after calling `search`. It behaves exactly the same.)

**The problem occurs when the search is cleared** (== search input = empty). THEN I still would like to scroll to the still selected element. However, jstree closes ALL previously opened nodes restoring the original opened state BUT does not allow me to re-open the branch of nodes where the **still selected** node resides.


In contrast, when reloading the page, the selected node is still selected and its parent branch is opened as expected. It's clearing the search that causes the issue.

**Comment by vakata** [comment]
https://www.jstree.com/api/#/?q=close_opene&f=$.jstree.defaults.search.close_opened_onclear - set this search plugin config option to `false`.

**Comment by srkunze** [comment]
> https://www.jstree.com/api/#/?q=close_opene&f=$.jstree.defaults.search.close_opened_onclear - set this search plugin config option to false.

Unfortunately, it is not intended to leave almost all nodes open on clear (user starts searching by typing the letter 'e'). So, close_opened_onclear=true is the right choice for me.

Looking at https://github.com/vakata/jstree/blob/master/dist/jstree.js#L3132 I found that select_node does indeed not re-open all parents of a node if that very node is already selected.

So, I went for the internal method _open_to. Looking at select_node, I realized, I need to pass the node object, not the ID. So, my solution looks like this:

tree._open_to(tree.get_node(tree.get_node(SPECIAL_ID)));


I don't really like internal APIs. So, would it be possible to add another settings to search? Basically, that leaves just the paths to all selected nodes open when clearing the search?

search.leave_selected_open_onclear=true would work for me thaen.
